presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestExpressionRewriteRuleSet.java

```diff
@@ -0,0 +1,139 @@
+                        a.globalGrouping()
+                                .addAggregation(
+                                        p.symbol("count_1", BigintType.BIGINT),
+                                        new FunctionCall(QualifiedName.of("count"), ImmutableList.of()),
+                                        ImmutableList.of(BigintType.BIGINT))
+                                .source(
+                                        p.values())))
+                .matches(
+                        PlanMatchPattern.aggregation(
+                                ImmutableMap.of("count_1", functionCall("none", ImmutableList.of())),
+                                PlanMatchPattern.values()));
+    }
+
+    @Test
+    public void testFilterExpressionRewrite()
+    {
+        tester().assertThat(zeroRewriter).on(p ->
+                p.filter(new LongLiteral("1"), p.values()))
+                .matches(
+                        filter("0", PlanMatchPattern.values()));
+    }
+
+    @Test
+    public void testValueExpressionRewrite()
+    {
+        tester().assertThat(zeroRewriter).on(p ->
+                p.values(ImmutableList.<Symbol>of(p.symbol("a")), ImmutableList.of((ImmutableList.of(PlanBuilder.expression("1"))))))
+                .matches(
+                        values(ImmutableList.of("a"), ImmutableList.of(ImmutableList.of(new LongLiteral("0")))));
+    }
+
+    @Test
+    public void testApplyExpressionRewrite()
+    {
+        tester().assertThat(new ApplyExpressionRewriter()).on(p ->
+                p.apply(
+                        Assignments.of(
+                                p.symbol("a", BigintType.BIGINT),
+                                new InPredicate(
+                                        new LongLiteral("0"),
+                                        new InListExpression(ImmutableList.of(
+                                                new LongLiteral("1"),
+                                                new LongLiteral("2"))))),
+                        ImmutableList.of(),
+                        p.values(),
+                        p.values()))
+                .matches(
+                        apply(
+                                ImmutableList.of(),
+                                ImmutableMap.of("a", expression("1 IN (1, 2)")),
+                                values(),
+                                values()));
+    }
+
+    private static class ZeroExpressionRewriter
+            extends ExpressionRewriteRuleSet
+    {
+        private ZeroExpressionRewriter() {super((expression, context) -> new LongLiteral("0"));}
+    }
+
+    private static class FunctionCallExpressionRewriter
+            extends ExpressionRewriteRuleSet
+    {
+        private static final FunctionCall noneCall = new FunctionCall(QualifiedName.of("none"), ImmutableList.of());
+
+        private FunctionCallExpressionRewriter() {super((expression, context) -> noneCall);}
+    }
+
+    private static class ApplyExpressionRewriter
```

Comment:
I think you don't need to create any inner class here as `ExpressionRewriteRuleSet` is not abstract, you can just return its instance with a proper lambda injected